src/AccessDefinitions/Metadata/AccessDefinitionClassMetadata.php

```diff
@@ -95,21 +75,45 @@ public function getPropertyMetadata(string $propertyName): ?AccessDefinitionProp
    * @return $this
    */
   public function setClassMetadataFromConfigData(array $data): self {
```

Comment:
Can we make this method more generic? Seems like we're nearing a point where maintainability becomes an issue.

Reply:
Not sure what you mean by "more generic". But I do agree that the readability of this method is not great.